src/box/lua/console.lua

```diff
@@ -985,7 +986,14 @@ local function client_handler(client, _peer)
     state:print(string.format("%-63s\n%-63s\n",
         "Tarantool ".. version.." (Lua console)",
         "type 'help' for interactive help"))
+    local on_shutdown = function()
+        state.running = false
+        fiber.cancel(state.fiber)
```

Comment:
Cancelling the console fiber will raise an exception in it so setting `self.running` here to false and checking it in `function repl()` looks pointless. Also, it `repl` fails with an exception, then `box.ctl.on_shutdown` trigger won't be reset.

Reply:
Changed to close socket to peep instead as dicussed f2f.

New version of console patch:

```diff
--- a/src/box/lua/console.lua
+++ b/src/box/lua/console.lua
@@ -985,7 +985,15 @@ local function client_handler(client, _peer)
     state:print(string.format("%-63s\n%-63s\n",
         "Tarantool ".. version.." (Lua console)",
         "type 'help' for interactive help"))
+    local on_shutdown = function()
+        -- Fiber is going to be cancelled on shutdown. Do not report
+        -- cancel induced error to the peer.
+        client:close();
+    end
+    state.fiber = fiber.self()
+    box.ctl.on_shutdown(on_shutdown)
     repl(state)
+    box.ctl.on_shutdown(nil, on_shutdown)
     session_internal.run_on_disconnect()
 end
```